db/migrate/20240515215452_add_marketplace_id_to_tags.rb

```diff
@@ -0,0 +1,7 @@
+class AddMarketplaceIdToTags < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    add_reference :marketplace_tags, :marketplace, index: {unique: true, algorithm: :concurrently}
```

Comment:
We probably don't want to have any empty foreign keys in this table, so it would be good to add that constraint as you add the reference.

This will also force you to consider how do you want to deal with any tag records that might already exist in the database (assign them to a random marketplace from the bazaar they belonged to? delete them? something else?).